proposals/lip-0053.md

```diff
@@ -133,28 +144,247 @@ The following constants are used throughout the document:
+| `MerkleRoot` | bytes   | Must be of length `MERKLE_ROOT_LENGTH`. | The root of a Merkle tree.   |
+| `PublicKeyEd25519` | bytes | Must be of length `ED25519_PUBLIC_KEY_LENGTH`. | Used for Ed25519 public keys. |
+| `PublicKeyBLS` | bytes | Must be of length `BLS_PUBLIC_KEY_LENGTH`. | Used for BLS keys. |
+| `ActiveValidator` | object | Must have `blsKey: PublicKeyBLS` and `bftWeight: uint64` properties. | The BLS key and BFT weight of an active validator. |
+
+### Auxiliary Functions
+
+The following auxiliary functions are used internally by the Interoperability module to verify and execute cross-chain updates.
+
+#### verifyLivenessCondition
+
+```python
+def verifyLivenessCondition(trs: Transaction) -> None:
+    sidechainID = trs.params.sendingChainID
+
+    if chainAccount(sidechainID).status == CHAIN_STATUS_REGISTERED:
+        certificateBytes = trs.params.certificate
+        if len(certificateBytes) == 0:
+            raise Exception("The first mainchain CCU must contain a non-empty certificate.")
+        
+        certificate = decode(certificateSchema, certificateBytes)
+        timestamp = timestamp of the block including CCU
+        if len(trs.params.inboxUpdate) > 0 and timestamp - certificate.timestamp > LIVENESS_LIMIT / 2:
+            raise Exception(f"First certificate cannot be older than {LIVENESS_LIMIT / 2} seconds.")
+
+    elif not isLive(sidechainID):
+        raise Exception("Sidechain is not live.")
+``` 
+
+#### verifyCertificate
+
+```python
+def verifyCertificate(trs: Transaction) -> None:
+    certificate = decode(certificateSchema, trs.params.certificate)
+    partnerchainID = trs.params.sendingChainID
+    partnerchainAccount = chainAccount(partnerchainID)
+    # Certificate height must be greater than last certificate height.
+    if certificate.height <= partnerchainAccount.lastCertificate.height:
+        raise Exception("Certificate height is not greater than last certificate height.")
+    # Certificate timestamp must be smaller than timestamp of the block including the CCU.
+    timestamp = timestamp of the block including CCU
+    if certificate.timestamp >= timestamp:
+        raise Exception("Certificate timestamp is not smaller than timestamp of the block including the CCU.")
+```
+
+#### verifyCertificateSignature
+
+```python
+def verifyCertificateSignature(trs: Transaction) -> None:
+    # Certificate signature must be a valid aggregate signature for the sdechain validators.
+    # verifyAggregateCertificateSignature is defined in LIP 0061.
+    certificate = decode(certificateSchema, trs.params.certificate)
+    partnerchainID = trs.params.sendingChainID
+    partnerchainAccount = chainAccount(partnerchainID)
+    validatorKeys = [validator.blsKey for validator in validators(partnerchainID).activeValidators]
+    validatorWeights = [validator.bftWeight for validator in validators(partnerchainID).activeValidators]
+    verifyAggregateCertificateSignature(
```

Comment:
It would be nice to link https://github.com/LiskHQ/lips/blob/main/proposals/lip-0061.md#verifyaggregatecertificatesignature so it is clear where this function is defined.

Reply:
added